src/pint/fitter.py

```diff
@@ -1181,7 +1181,9 @@ def fit_toas(self, maxiter=1, threshold=0, full_cov=False):
                 fitpv[pn] = np.longdouble((pv + dpv) / fitp[pn].units)
                 # NOTE We need some way to use the parameter limits.
                 fitperrs[pn] = errs[uind]
-            self.minimize_func(list(fitpv.values()), *list(fitp.keys()))
+            newparams = dict(zip(list(fitp.keys()), list(fitpv.values())))
```

Comment:
Why this change?

Reply:
It's more transparent, but more importantly, within minimize_func, the Residuals get re-instantiated, so self.resids._chi2 gets reset to None, so the return statement of that function winds up calling calc_chi2, which is not necessary (especially given this dev) and time consuming.

Reply:
Does this behave any differently? It seems like both call `set_params` followed by `update_resids`? (One could check using `breakpoint()`)

Reply:
Yes it does; update_resids resets the residuals in the same way, however there's no querying for self.resids.chi2 at the end, which is the return statement of minimize_func (since it's purpose is to return a chi2, which we don't need here anyway) and which calls calc_chi2 because self.resids.chi2 at that point is None.  This fix makes the updating of resids and pars transparent, and then a few lines later it explicitly sets self.resids.chi2 to the end result of the fit, it doesn't rely on calc_chi2 at all.